### PR TITLE
Issue #218: mail system settings

### LINKED ISSUE #218: Mail system settings
Installing ubercart produces the following settings in system.mail
`    "mail_system": {
        "default-system": "MimeMailSystem",
        "uc_cart": "UbercartMailSystem",
        "uc_order": "UbercartMailSystem",
        "uc_file": "UbercartMailSystem",
        "uc_roles": "UbercartMailSystem",
        "uc_stock": "UbercartMailSystem",
        "uc_store": "UbercartMailSystem"
    }`
This does not seem to be right but before suggesting a fix I would like to know what is the intention of this? Does Ubercart need different mail systems to be selectable for its different components or is it intended for different theming of emails from different companents? Please would someone explain.

### PR DESCRIPTION
**Pull request by Graham-72**
This PR inserts the necessary settings into core's system.mail config file and removes them if Ubercart's Store module is uninstalled.

### CODE REVIEW

**Comment by herbdool** [comment]
@Graham-72 you also need an update hook to change people's settings for those who have already installed this.

**Comment by Graham-72** [comment]
@herbdool thanks for the prompt on this. I think this addition to the PR will do?

File: uc_store/uc_store.install [hunk 1]
```diff
@@ -218,6 +224,21 @@ function uc_store_update_last_removed() {
 /**
  * Implements hook_update_N().
```
Comment (herbdool) [inline uc_store/uc_store.install]: I recommend making the function message more descriptive of what the update is doing. This gets parsed and displayed for the end user. See how core does it for an example: https://github.com/backdrop/backdrop/blob/7572f3d41e157a04b262ee262884b6d463ef0ae2/core/modules/contact/contact.install#L54

File: uc_store/uc_store.install [hunk 2]
```diff
@@ -218,6 +224,21 @@ function uc_store_update_last_removed() {
...
  */
+function uc_store_update_1001() {
```
Comment (herbdool) [inline uc_store/uc_store.install]: Also, please put this update hook at the bottom, in numerical order. Right now it was pasted *before* uc_store_update_1000.

File: uc_store/uc_store.install (hunk 1 above)
Comment (Graham-72) [reply uc_store/uc_store.install]: Function message for 1000 changed and for 1001 added.

File: uc_store/uc_store.install (hunk 2 above)
Comment (Graham-72) [reply uc_store/uc_store.install]: Update hook moved to bottom.

**Comment by herbdool** [comment]
@Graham-72 looks good now.

Note I haven't tested that the actual configuration is correct, but seems like it was working for you.

**Pull request merged**